os/kernel/log_dump/log_dump.c

```diff
@@ -125,33 +142,39 @@ static void log_dump_mem_check(size_t max_size)
 	}
 }
 
-int log_dump_save(char ch)
+int log_dump_tobuffer(char ch, size_t *free_size)
 {
 	/* need to check if the current chunks size is over max_log_size or greater than x% of free heap */
 
-	size_t free_size;
+	size_t max_size;
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	free_size = mm_get_heap_free_size(void);
+	*free_size = mm_get_heap_free_size();
```

Comment:
Could you change the `mm_get_heap_free_size` to `kmm_xxx`?
You can refer to this.
https://github.com/Samsung/TizenRT/commit/0fbdf817f843109d469bf3af5fa3a7bc04edc05b